content/actions/using-workflows/workflow-commands-for-github-actions.md

```diff
@@ -412,6 +408,191 @@ jobs:
 
 {% endpowershell %}
 
+### Example: Masking a generated output within a single job
+
+{% ifversion actions-save-state-set-output-envs %}
+{% else %}
+{% note %}
+**Note**: You must use `add-mask` before you use `set-output`. Otherwise, the output will not be masked.
+{% endnote %}
+{% endif %}
+
+If you do not need to pass your secret from one job to another job, you can:
+1. Generate the secret (without outputting it).
+1. Mask it with `add-mask`.
+{% ifversion actions-save-state-set-output-envs %}
+1. Use `GITHUB_OUTPUT` to make the secret available to other steps within the job.
+{% else %}
+1. Use `set-output` to make the secret available to other steps within the job.
+{% endif %}
+
+{% bash %}
+
+```yaml{:copy}
+on: push
+jobs:
+  generate-a-secret-output:
+    runs-on: ubuntu-latest
+    steps:
+      - id: sets-a-secret
+        name: Generate, mask, and output a secret
+        run: |
+          the_secret=$((RANDOM))
+          echo "::add-mask::$the_secret"
+{% ifversion actions-save-state-set-output-envs %}
+          echo "secret-number=$the_secret" >> "$GITHUB_OUTPUT"
+{% else %}
+          echo "::set-output name=secret-number::$the_secret"
+{% endif %}
+      - name: Use that secret output (protected by a mask)
+        run: |
+{% raw %}
+          echo "the secret number is ${{ steps.sets-a-secret.outputs.secret-number }}"
+{% endraw %}
+```
+
+{% endbash %}
+
+{% powershell %}
+
+```yaml{:copy}
+on: push
+jobs:
+  generate-a-secret-output:
+    runs-on: ubuntu-latest
+    steps:
+      - id: sets-a-secret
+        name: Generate, mask, and output a secret
+        shell: pwsh
+        run: |
+          Set-Variable -Name TheSecret -Value (Get-Random)
+          Write-Output "::add-mask::$TheSecret"
+{% ifversion actions-save-state-set-output-envs %}
+          "secret-number=$TheSecret" >> $env:GITHUB_OUTPUT
+{% else %}
+          Write-Output "::set-output name=secret-number::$TheSecret"
+{% endif %}
+      - name: Use that secret output (protected by a mask)
+        shell: pwsh
+        run: |
+{% raw %}
+          Write-Output "the secret number is ${{ steps.sets-a-secret.outputs.secret-number }}"
+{% endraw %}
```

Comment:
<img width="765" alt="image" src="https://user-images.githubusercontent.com/2119212/219968950-fb49bcd3-2464-4b23-86b5-108f011f514b.png">

I don't want to think about the blank lines. I'm sure I could get rid of them by merging everything into a single line, but that's really unreadable.

https://github.com/jsoref/docs-19886-926e50/actions/runs/4217658308/jobs/7321598494
<img width="827" alt="image" src="https://user-images.githubusercontent.com/2119212/219969001-b2f46f77-a55b-4d79-a9f4-b69e4efa5ea3.png">

```
2s
Run Set-Variable -Name TheSecret -Value (Get-Random)
  Set-Variable -Name TheSecret -Value (Get-Random)
  Write-Output "::add-mask::$TheSecret"
  
  Write-Output "secret-number=$TheSecret" >> "$env:GITHUB_OUTPUT"
  shell: /usr/bin/pwsh -command ". '{0}'"
2s
Run Write-Output "the secret number is ***"
  
  Write-Output "the secret number is ***"
  shell: /usr/bin/pwsh -command ". '{0}'"
the secret number is ***
```